examples/linalg/src/linalg.rs

```diff
@@ -0,0 +1,43 @@
+use hacspec_lib::*;
+
+pub type DimType = usize;
+pub type Scalar = i128;
+pub type Dims = (DimType, DimType);
+pub type Matrix = (Dims, Seq<Scalar>);
+
+// Generate new matrix using rows, cols and a seq.
+pub fn new(rows: DimType, cols: DimType, seq: Seq<Scalar>) -> Result<Matrix, ()> {
+    if seq.len() > 0 && rows * cols == seq.len() {
+        Result::<Matrix, ()>::Ok(((rows, cols), seq))
```

Comment:
You can define a `type` for `Result::<Matrix, ()>` to make this look a little nicer.